compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1984,43 +2026,10 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         || symInfo.isInstanceOf[MethodType]
             && symInfo.signature.consistentParams(info2.signature)
 
-      def tp1IsSingleton: Boolean = tp1.isInstanceOf[SingletonType]
-
-      // A relaxed version of isSubType, which compares method types
-      // under the standard arrow rule which is contravariant in the parameter types,
-      // but under the condition that signatures might have to match (see sigsOK)
-      // This relaxed version is needed to correctly compare dependent function types.
-      // See pos/i12211.scala.
-      def isSubInfo(info1: Type, info2: Type, symInfo1: Type): Boolean =
-        def fallback = inFrozenGadtIf(tp1IsSingleton) { isSubType(info1, info2) }
-        info2 match
-          case info2: PolyType if ccEnabled => // See TODO about `ccEnabled` in `sigsOK`, this should also go under `relaxedSubtyping`.
-            info1 match
-              case info1: PolyType =>
-                comparingTypeLambdas(info1, info2):
-                  info1.paramNames.hasSameLengthAs(info2.paramNames)
-                  && isSubInfo(info1.resultType, info2.resultType.subst(info2, info1), symInfo1.resultType)
-                  // Signature checks are never necessary because polymorphic
-                  // refinements are only allowed for the `apply` method of a
-                  // PolyFunction.
-              case _  => fallback
-          case info2: MethodType =>
-            info1 match
-              case info1: MethodType =>
-                matchingMethodParams(info1, info2, precise = false)
-                && isSubInfo(info1.resultType, info2.resultType.subst(info2, info1), symInfo1.resultType)
-                && sigsOK(symInfo1, info2)
-              case _ => fallback
-          case info2 @ CapturingType(parent2, refs2) if ctx.phase == Phases.checkCapturesPhase =>
-            val refs1 = info1.captureSet
-            (refs1.isAlwaysEmpty || subCaptures(refs1, refs2, frozenConstraint).isOK) && sameBoxed(info1, info2, refs1)
-              && isSubInfo(info1, parent2, symInfo1)
-          case _ =>
-            info1 match
-              case info1 @ CapturingType(parent1, refs1) if ctx.phase == Phases.checkCapturesPhase =>
-                subCaptures(refs1, info2.captureSet, frozenConstraint).isOK && sameBoxed(info1, info2, refs1)
-                  && isSubInfo(parent1, info2, symInfo1)
-              case _ => fallback
+      def fallback = (info1: Type, info2: Type) =>
+        def tp1IsSingleton: Boolean = tp1.isInstanceOf[SingletonType]
```

Comment:
Using a def here instead of a val doesn't seem useful since it's evaluated in the next line